plotting/plot_factory/plot_factory.py

```diff
@@ -0,0 +1,258 @@
+# ############################################################################### #
+# Autoreduction Repository : https://github.com/ISISScientificComputing/autoreduce
+#
+# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI
+# SPDX - License - Identifier: GPL-3.0-or-later
+# ############################################################################### #
+"""
+Constructs a plot and DashApp object for insertion into directly into a web page
+"""
+
+# TODO: Allow for mode and plot to not exist in yaml file and plot to still be displayed.
+# Note pandas and plotly.graph_objs dependencies are used, but not recognised by pycharm interpreter
+
+# Visualisation Dependencies
+import dash
+import dash_core_components as dcc
+import dash_html_components as html
+import plotly.graph_objects as go
+
+# Internal Dependencies
+from plotting.plot_meta_language.interpreter import Interpreter
+
+
+# Data Dependencies
+
+
+class PlotFactory:
+    """Creates figures from formatted data and layout
+       producing a DashApp for direct insertion inside a given web page
+       =_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=
+        """
+
+    @staticmethod
+    def get_trace_list(data, layout, figure_name):
+        """Creates trace list containing traces for each spectrum to place in figure
```

Comment:
Can we change these to align with the docstring style of the rest of the code base. Whilst this style has advantages i think given the fact we are aiming for many small methods, it means the vast majority of this file is doc strings. 
We might consider moving to this in the future but I think we'll do it as a separate project in of itself. For the time being lets be consistent with the rest of the code base